### PR TITLE
Add PLS, a language server for Perl

### PR DESCRIPTION
**Pull request by WhoIsSethDaniel**

### CODE REVIEW

File: lua/lspconfig/perlpls.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  default_config = {
+    cmd = {"pls"};
+    settings = {
+      perl = {
+        perlcritic = { enabled = false };
+      };
+    };
+    filetypes = {"perl"};
+    root_dir = function(fname)
+      return util.root_pattern(".git")(fname) or vim.fn.getcwd()
```
Comment (mjlbach) [inline lua/lspconfig/perlpls.lua:14]: use util.path.dirname instead of cwd, see pyright

File: lua/lspconfig/perlpls.lua [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    end;
+  };
+  docs = {
+    package_json = "https://raw.githubusercontent.com/FractalBoy/perl-language-server/master/client/package.json";
+    description = [[
+https://github.com/FractalBoy/perl-language-server
```
Comment (mjlbach) [inline lua/lspconfig/perlpls.lua:20]: Can you add more description on how to install this? see pyright

**Comment by mjlbach** [comment]
Please rebase your PR onto the latest master branch (it includes new linting checks)

File: lua/lspconfig/perlpls.lua (hunk 2 above)
Comment (WhoIsSethDaniel) [reply lua/lspconfig/perlpls.lua:20]: I'm not the author of PLS so I don't have write access to that github page. I am working with the author to get Neovim instructions included. Part of that process is to get this configuration included in nvim-lspconfig.

Comment (mjlbach) [reply lua/lspconfig/perlpls.lua:20]: Oh no worries, we usually add a small snippet here as well

File: lua/lspconfig/perlpls.lua (hunk 1 above)
Comment (WhoIsSethDaniel) [reply lua/lspconfig/perlpls.lua]: done.

**Comment by WhoIsSethDaniel** [comment]
> Please rebase your PR onto the latest master branch (it includes new linting checks)

done.

**Comment by mjlbach** [comment]
Some lint errors, you can directly run https://github.com/JohnnyMorganz/StyLua from the root of nvim-lspconfig with the path to your file to automatically fix them

**Comment by WhoIsSethDaniel** [comment]
> Some lint errors, you can directly run https://github.com/JohnnyMorganz/StyLua from the root of nvim-lspconfig with the path to your file to automatically fix them

Thanks.  Did this. Seems to have passed all checks.

**Pull request merged**